src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -1221,13 +1264,20 @@ fn load_stratis_metadata(
             )));
     }
 
-    match get_metadata(&infos) {
-        Ok(opt) => opt
+    match get_metadata(&stratis_infos_ref(infos)) {
+        Ok(opt) => {
+            let (timestamp, mut metadata) = opt
             .ok_or_else(|| {
                 StratisError::Msg(format!(
                     "No metadata found on devices associated with pool UUID {pool_uuid}"
                 ))
-            }),
+            })?;
+            if remove_cache {
+                let dev_uuids = remove_cache_from_metadata(&mut metadata);
+                update_infos(infos, dev_uuids);
+            }
+            Ok((timestamp, metadata))
```

Comment:
Should the timestamp be updated here to the current time?

Reply:
I think this would be a bad idea, only because we use timestamp throughout our entire codebase when reading the metadata. If you want to put up a separate PR changing all of the instances of timestamp to last_updated_time, we can but I think that should be separate from this PR.